src/main/java/org/jpeek/web/Reports.java

```diff
@@ -151,6 +130,34 @@
         return new TypedPages(new Pages(output));
     }
 
+    /**
+     * Extract classes from passed Jar file.
+     * @param origin File name
+     * @param dest Destination dir
+     * @throws IOException If fails
+     */
+    private static void extractClasses(final String origin, final String dest) throws IOException {
+        final JarFile jar = new JarFile(origin);
+        final Enumeration<JarEntry> entries = jar.entries();
+        while (entries.hasMoreElements()) {
+            final JarEntry entry = entries.nextElement();
+            final File item = new File(dest + File.separator + entry.getName());
+            if (entry.isDirectory()) {
+                item.mkdir();
+                continue;
+            }
+            try (
```

Comment:
@dgarus I think you can leverage `Cactoos `io package to reduce the below code.
Look at the Cactoos `TeeInput`